[AC-1163] Manage users permissions should allow collection assignment

shane-melton opened the pull request:
## Type of change

<!-- (mark with an `X`) -->

```
- [X] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective
Update the list collections endpoint to include all organization collections if the requesting user has the `ManageUsers` permission.

Users with the `ManageUsers` permission need to be able to view all collections in order to assign them to other users.


## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

eliykat reviewed (commented):
I'm not sure about the consequences of this change - I think "ViewAllCollections" can refer to the _contents_ of collections (i.e. ciphers), not just the collection objects themselves. e.g. `CiphersController.GetAdmin` seems to treat it this way. Is it possible to make this change in a more targeted way in the controller (or otherwise)?

shane-melton commented:
@eliykat Great point; I didn't realize `ViewAllCollections` was being used to view the collection contents in other locations. I updated the change to target the controller endpoint specifically as suggested and reverted the change to `ViewAllCollections`.

eliykat reviewed (approved):
Much safer, thanks!
